SUL23-40 | @jdwjdwjdw | Add styles for News nodes, lists, and cards

jdwjdwjdw opened the pull request:
# READY FOR REVIEW

# Summary
- [SUL23-40](https://stanfordits.atlassian.net/browse/SUL23-40): FE-DEV | Build "News" content display in NextJS
- Add styles for News nodes, lists, and cards
- Add social icon components for Facebook, Twitter, and linkedin 
- Add additional `footer` field to Card component

# Review By (Date)
- When does this need to be reviewed by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Build out some News pages
2. Confirm the individual news node pages match D9
3. Build default News lists and card grid News list
4. Confirm they match D9

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)


<a href="https://gitpod.io/#https://github.com/SU-SWS/sulgryphon-nextjs/pull/24"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -20,27 +25,63 @@ export const NodeStanfordNews = ({node, ...props}: NewsNodeProps) => {
+          <div className="su-flex md:su-order-last su-rs-mb-2">
+            <ul className="su-flex su-list-unstyled md:su-pl-[10px] su-mt-[-3px]">
+              <li className="su-mr-1em">
+                <DrupalLink className="su-text-black hocus:su-text-facebook su-transition-colors" href="https://www.facebook.com">
```
This should be a share link like `http://www.facebook.com/sharer.php?u=[url]&display=popup`

pookmish replied:
Also, feel free to just use `<Link>`. it does the same thing as `<DrupalLink>` unless you need a button.

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -20,27 +25,63 @@ export const NodeStanfordNews = ({node, ...props}: NewsNodeProps) => {
+                </DrupalLink>
+              </li>
+              <li className="su-mr-1em">
+                <DrupalLink className="su-text-black hocus:su-text-twitter su-transition-colors" href="https://twitter.com">
```
This should be a share link like `https://twitter.com/intent/tweet?url=[url]`

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -20,27 +25,63 @@ export const NodeStanfordNews = ({node, ...props}: NewsNodeProps) => {
+                </DrupalLink>
+              </li>
+              <li className="su-mr-1em">
+                <DrupalLink className="su-text-black hocus:su-text-linkedin su-transition-colors" href="https://www.linkedin.com/">
```
This should be a share link like `https://www.linkedin.com/shareArticle?mini=true&url=[url]`

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -20,27 +25,63 @@ export const NodeStanfordNews = ({node, ...props}: NewsNodeProps) => {
+                </DrupalLink>
+              </li>
+              <li className="su-mr-1em">
+                <DrupalLink className="su-text-black  hocus:su-text-digital-blue su-transition-colors" href="/">
```
This should be a mailto link like `mailto:?subject=[title]body=[url]`

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -20,27 +25,63 @@ export const NodeStanfordNews = ({node, ...props}: NewsNodeProps) => {
+                </DrupalLink>
+              </li>
+              <li className="su-mr-1em">
+                <DrupalLink className="su-text-black  hocus:su-text-digital-blue su-transition-colors" href="/">
```
This should have a print `onclick` like `onClick="window.print();return false;"`

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -20,27 +25,63 @@ export const NodeStanfordNews = ({node, ...props}: NewsNodeProps) => {
-      <article {...props}>
+      <article {...props} className="su-mt-50">
         {node.su_news_topics && node.su_news_topics.map(topic =>
           <div key={topic.id}>
```
This should be inline with each other with a `,` between them.

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -59,38 +100,69 @@ export const NodeStanfordNews = ({node, ...props}: NewsNodeProps) => {
-        {node.su_news_topics && node.su_news_topics.map(topic =>
+    <article {...props}>
+      <Card
+        image={
```
`image` should be null if no image is available. Verify you don't get an empty space in the cards. You might take the basic page as an example: https://github.com/SU-SWS/sulgryphon-nextjs/blob/1.x/components/nodes/node-stanford-page.tsx#L88

pookmish commented on components/patterns/card.tsx:
```diff
@@ -45,6 +46,10 @@ export const Card = ({video, image, superHeader, header, body, link, linkStyle,
         {body && <div>{formatHtml(body)}</div>}
 
+        {footer &&
+            <h3 className="su-leading-display su-text-18 su-rs-pt-0 su-text-digital-red su-font-normal">{footer}</h3>
```
are you sure it should be an `h3` element?

pookmish commented on components/simple/icons/FacebookIcon.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
It doesn't look like you need this import

pookmish reviewed (changes requested):
Mostly just some minimal things. Overall it's looking really good. I created some test content on the Acquia site which are getting pulled over:
https://deploy-preview-24--stanford-library.netlify.app/news/sample-stanford-researchers-find-misfiring-jittery-neurons
https://deploy-preview-24--stanford-library.netlify.app/test-basic-page-content/news-content

jdwjdwjdw commented:
Thanks @pookmish! I think I addressed everything you mentioned. Also fyi - I added 'decanter/fonts.css` to our styles, which should fix up some things (e.g. the Stanford logo in our footer).

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -1,46 +1,104 @@
+                </Link>
+              </li>
+              <li className="su-mr-1em">
+                <Link className="su-text-black hocus:su-text-twitter su-transition-colors" href={`https://twitter.com/intent/tweet?url={window.location}&text=${currentUrl}`}>
```
Shouldn't url be the `currentUrl` and the 5ext be the title?

jdwjdwjdw replied:
ah good catch, thanks

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -1,46 +1,104 @@
+              <li className="su-mr-1em">
+                <Link className="su-text-black  hocus:su-text-digital-blue su-transition-colors" href={`mailto:?subject=${node.title}&body=${currentUrl}`}>
+                  <span className="su-sr-only">Forward Email</span>
+                  <EnvelopeIcon className="su-w-[28px]"/>
```
You can use the `width` property instead of css. It should work the same.

pookmish commented on components/nodes/node-stanford-news.tsx:
```diff
@@ -1,46 +1,104 @@
+              <li className="su-mr-1em">
+                <button onClick={() => window.print()} className="su-text-black  hocus:su-text-digital-blue su-transition-colors">
+                  <span className="su-sr-only">Print Article</span>
+                  <PrinterIcon className="su-w-[28px]"/>
```
Same as envelope icon

pookmish reviewed (changes requested):
Just a couple last tweaks and I'll merge it in

jdwjdwjdw commented:
@pookmish should be good to go now

pookmish commented:
Thanks. Looks good
